stackit/internal/services/serviceaccount/account/resource.go

```diff
@@ -0,0 +1,363 @@
+	return &serviceAccountResource{}
+}
+
+// serviceAccountResource implements the resource interface for service accounts.
+type serviceAccountResource struct {
+	client *serviceaccount.APIClient
+}
+
+// Configure sets up the API client for the service account resource.
+func (r *serviceAccountResource) Configure(ctx context.Context, req resource.ConfigureRequest, resp *resource.ConfigureResponse) {
+	// Prevent potential panics if the provider is not properly configured.
+	if req.ProviderData == nil {
+		return
+	}
+
+	// Validate provider data type before proceeding.
+	providerData, ok := req.ProviderData.(core.ProviderData)
+	if !ok {
+		core.LogAndAddError(ctx, &resp.Diagnostics, "Error configuring API client", fmt.Sprintf("Expected configure type stackit.ProviderData, got %T", req.ProviderData))
+		return
+	}
+
+	if !resourceBetaCheckDone {
+		features.CheckBetaResourcesEnabled(ctx, &providerData, &resp.Diagnostics, "stackit_service_account", "resource")
+		if resp.Diagnostics.HasError() {
+			return
+		}
+		resourceBetaCheckDone = true
+	}
+
+	// Initialize the API client with the appropriate authentication and endpoint settings.
+	var apiClient *serviceaccount.APIClient
+	var err error
+	if providerData.ServiceAccountCustomEndpoint != "" {
+		apiClient, err = serviceaccount.NewAPIClient(
+			config.WithCustomAuth(providerData.RoundTripper),
+			config.WithEndpoint(providerData.ServiceAccountCustomEndpoint),
+		)
+	} else {
+		apiClient, err = serviceaccount.NewAPIClient(
+			config.WithCustomAuth(providerData.RoundTripper),
+		)
+	}
+
+	// Handle API client initialization errors.
+	if err != nil {
+		core.LogAndAddError(ctx, &resp.Diagnostics, "Error configuring API client", fmt.Sprintf("Configuring client: %v. This is an error related to the provider configuration, not to the resource configuration", err))
+		return
+	}
+
+	// Store the initialized client.
+	r.client = apiClient
+	tflog.Info(ctx, "Service Account client configured")
+}
+
+// Metadata sets the resource type name for the service account resource.
+func (r *serviceAccountResource) Metadata(_ context.Context, req resource.MetadataRequest, resp *resource.MetadataResponse) {
+	resp.TypeName = req.ProviderTypeName + "_service_account"
+}
+
+// Schema defines the schema for the resource.
+func (r *serviceAccountResource) Schema(_ context.Context, _ resource.SchemaRequest, resp *resource.SchemaResponse) {
+	descriptions := map[string]string{
+		"id":         "Terraform's internal resource ID, structured as \"project_id,email\".",
+		"project_id": "STACKIT project ID to which the service account is associated.",
+		"name":       "Name of the service account.",
+		"email":      "Email of the service account.",
+	}
+
+	resp.Schema = schema.Schema{
+		MarkdownDescription: features.AddBetaDescription("Schema for a STACKIT service account resource."),
+		Description:         "Schema for a STACKIT service account resource.",
+		Attributes: map[string]schema.Attribute{
```

Comment:
More consistent to the existing descriptions


```suggestion
		MarkdownDescription: features.AddBetaDescription("Service account resource schema."),
		Description:         "Service account resource schema.",
		Attributes: map[string]schema.Attribute{
```